### PR TITLE
[Refactor] fix sonar: limit the size of left shift

### PR DESCRIPTION
**Pull request by fzhedu**
Signed-off-by: fzhedu <fzhedu@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes # https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=BUG&id=Starrocks_be&open=AYQ89hCYtI4mkxoJjYxS

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/util/sm3.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@ bool Sm3::is_little_endian = (*(char*)&Sm3::ENDIAN_TEST_VALUE == 1) ? true : fal
 
 // Cyclic shift left k-bit operation
 unsigned int Sm3::left_rotate(unsigned int word, int k) {
+    k = k % WORD_BITS;
```
Comment (imay) [inline be/src/util/sm3.cpp:11]: If I is greater than WORD BITS it should be it other than other number?

Comment (fzhedu) [reply be/src/util/sm3.cpp:11]: if k is greater or equal to the width of type 'unsigned int', the result of the right shift is undefined.

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Review by Pslydhh — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
run starrocks_be_unittest

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
@mergify backport branch-2.5

**Pull request merged**